server/fleet/activities.go

```diff
@@ -1682,6 +1682,7 @@ type ActivityAddedAppStoreApp struct {
 	TeamName      *string             `json:"team_name"`
 	TeamID        *uint               `json:"team_id"`
 	Platform      AppleDevicePlatform `json:"platform"`
+	SelfService   bool                `json:"self_service"`
```

Comment:
`ActivityTypeInstalledSoftware` seems to be done after the fact, so I'm not sure how I would make that association without immediately posting the activity, then possibly getting a duplicate later

Reply:
Oh nevermind, I see how this works. I can do that

Reply:
I'm not sure I understand, you don't need to change where it is created (note that I'm talking about `ActivityInstalledAppStoreApp`), it is already created here: https://github.com/fleetdm/fleet/blob/08763d59f1dd28c95ad133a7fbb666e27810ed3c/server/datastore/mysql/vpp.go#L527-L534

What I say is that I think this activity should also include a new `SelfService` field, but unlike the one you added in `ActivityAddedAppStoreApp` (which indicates whether or not the app CAN be self-installed), the one on `ActivityInstalledAppStoreApp` would indicate if the app WAS indeed self-installed. This would mirror what we do for software installers. I don't think it's more involved than reading the existing `self_service` column from `host_vpp_software_installs` here: https://github.com/fleetdm/fleet/blob/08763d59f1dd28c95ad133a7fbb666e27810ed3c/server/datastore/mysql/vpp.go#L459-L477 and passing it to the activity?

Reply:
Yes sorry, I took a closer look and understood how it works after leaving that comment.

Reply:
Done